### PR TITLE
Check status code before parsing response

### PR DESCRIPTION
**Pull request by sidraval**
Check that the status code is between 200 and 299 before parsing JSON. Return an appropriate error otherwise.

### CODE REVIEW

File: Source/Models/APIClient.swift [hunk 1]
```diff
@@ -20,8 +20,23 @@ public extension APIClient {
 }
 
 private func deserialize(response: HTTPResponse) -> Result<JSON, NSError> {
+  switch response.code {
+  case 200...299:
+    return parseJSON(response)
+  case 300...399:
+    return .Failure(Result<JSON, NSError>.error("Multiple choices: \(response.code)"))
```
Comment (gfontenot) [inline Source/Models/APIClient.swift:27]: Can you not omit the `Result<JSON, NSError>` bit here?

**Comment by gfontenot** [comment]
Makes sense to me. :+1:

**Comment by gfontenot** [comment]
@thoughtbot/ios any thoughts here?

**Comment by tonyd256** [comment]
looks good

**Comment by dazmuda** [comment]
Good stuff. 202!

**Comment by jnutting** [comment]
Good!

**Pull request merged**